packages/main/src/plugin/extension-loader.spec.ts

```diff
@@ -0,0 +1,224 @@
+
+const configurationRegistry: ConfigurationRegistry = {} as unknown as ConfigurationRegistry;
+
+const imageRegistry: ImageRegistry = {} as unknown as ImageRegistry;
+
+const apiSender: any = {};
+
+const trayMenuRegistry: TrayMenuRegistry = {} as unknown as TrayMenuRegistry;
+
+const dialogs: Dialogs = {} as unknown as Dialogs;
+
+const progress: ProgressImpl = {} as ProgressImpl;
+
+const notifications: NotificationImpl = {} as unknown as NotificationImpl;
+
+const statusBarRegistry: StatusBarRegistry = {} as unknown as StatusBarRegistry;
+
+const kubernetesClient: KubernetesClient = {} as unknown as KubernetesClient;
+
+const fileSystemMonitoring: FilesystemMonitoring = {} as unknown as FilesystemMonitoring;
+
+const proxy: Proxy = {} as unknown as Proxy;
+
+const containerProviderRegistry: ContainerProviderRegistry = {} as unknown as ContainerProviderRegistry;
+
+const inputQuickPickRegistry: InputQuickPickRegistry = {} as unknown as InputQuickPickRegistry;
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+beforeAll(() => {
+  extensionLoader = new TestExtensionLoader(
+    commandRegistry,
+    menuRegistry,
+    providerRegistry,
+    configurationRegistry,
+    imageRegistry,
+    apiSender,
+    trayMenuRegistry,
+    dialogs,
+    progress,
+    notifications,
+    statusBarRegistry,
+    kubernetesClient,
+    fileSystemMonitoring,
+    proxy,
+    containerProviderRegistry,
+    inputQuickPickRegistry,
+  );
+});
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+test('Should watch for files and load them at startup', async () => {
+  const fakeDirectory = '/fake/path/scanning';
+  const rootedFakeDirectory = path.resolve(fakeDirectory);
+
+  // fake scanning property
+  extensionLoader.setPluginsScanDirectory(fakeDirectory);
+
+  vi.mock('node:fs');
+  // mock fs.watch
+  const fsWatchMock = vi.spyOn(fs, 'watch');
+  fsWatchMock.mockReturnValue({} as fs.FSWatcher);
+
+  // mock fs.existsSync
+  const fsExistsSyncMock = vi.spyOn(fs, 'existsSync');
+  fsExistsSyncMock.mockReturnValue(true);
+
+  // mock fs.promises.readdir
+  const readdirMock = vi.spyOn(fs.promises, 'readdir');
+
+  const ent1 = {
+    isFile: () => true,
+    isDirectory: () => false,
+    name: 'foo.cdix',
+  } as unknown as fs.Dirent;
+
+  const ent2 = {
+    isFile: () => true,
+    isDirectory: () => false,
+    name: 'bar.foo',
+  } as unknown as fs.Dirent;
+
+  const ent3 = {
+    isFile: () => false,
+    isDirectory: () => true,
+    name: 'baz',
+  } as unknown as fs.Dirent;
+  readdirMock.mockResolvedValue([ent1, ent2, ent3]);
+
+  // mock loadPackagedFile
+  const loadPackagedFileMock = vi.spyOn(extensionLoader, 'loadPackagedFile');
+  loadPackagedFileMock.mockResolvedValue();
+
+  await extensionLoader.setupScanningDirectory();
+
+  // expect to load only one file (other are invalid files/folder)
+  expect(loadPackagedFileMock).toBeCalledWith(path.join(rootedFakeDirectory, 'foo.cdix'));
```

Comment:
```suggestion
  expect(loadPackagedFileMock).toBeCalledWith(path.join(fakeDirectory, 'foo.cdix'));
```

Fix Windows test failure